flink-table/flink-table-common/src/main/java/org/apache/flink/table/utils/DateTimeUtils.java

```diff
@@ -424,8 +424,12 @@ public static TimestampData parseTimestampData(String dateStr, int precision, Ti
     }
 
     public static TimestampData parseTimestampData(String dateStr, String format) {
-        DateTimeFormatter formatter = DATETIME_FORMATTER_CACHE.get(format);
-
```

Comment:
Address comment. https://github.com/apache/flink/pull/25763#discussion_r1893222879

Reply:
Nice!  This also makes sense to catch the IllegalArgumentException / parse errors!